kernel-plugins/src/BLSValidator.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.4 <0.9.0;
+
+import "kernel/src/validator/IValidator.sol";
+import "openzeppelin-contracts/contracts/utils/cryptography/EIP712.sol";
+import "kernel/src/utils/KernelHelper.sol";
+import "account-abstraction/samples/bls/lib/hubble-contracts/contracts/libs/BLS.sol";
+
+// BLSValidator is a validator that uses BLS signatures to validate transactions.
+// TODO: Consider account recovery, aggregate signatures, and use EIP 712.
+// Note: This is not audited and should not be used in production.
+contract BLSValidator is IKernelValidator {
+    bytes32 public constant BLS_DOMAIN = keccak256("eip4337.bls.domain");
+    
+    event OwnerChanged(address indexed kernel, uint256[4] indexed oldPublicKey, uint256[4] indexed newPublicKey);
+
+    mapping(address => uint256[4]) public blsValidatorStorage;
```

Comment:
What is this mapping for exactly? It has quite a generic name but appears to be a mapping of owners to bls public keys.

Is it to do with 4337 storage rules? https://docs.zerodev.app/extend-wallets/build-a-plugin#validator-storage

Reply:
It is a mapping of the address of the kernel wallet to a BLS public key.  So a transaction coming from that kernel wallet can be verified with the public key in the mapping.

Yeah not sure if this is how it is with `Safe` but the Validator plugins with kernel are link singletons in my mind. So several kernel wallets can use the same validator rather than a validator being deployed for each kernel wallet.  Acting similarly to our verification gateway.

But yeah is it to do with the storage rules, I copied the structure of the ECDSA validator that zerodev wrote.

Reply:
ok got it, that makes sense regarding the plugin as a singleton, and explains why a mapping would be used